### PR TITLE
[color addon] Add XYZ, xyY, Lab and LCH color spaces.

### PR DESCRIPTION
**Pull request by allefant**

### CODE REVIEW

File: addons/color/color.c [hunk 1]
```diff
@@ -575,4 +583,227 @@ uint32_t al_get_allegro_color_version(void)
...
+   return pow(x, 1 / 2.4) * (1 + a) - a;
+}
+
+
+/* Function: al_color_xyz_to_rgb
+ */
+void al_color_xyz_to_rgb(float x, float y, float z,
+    float *red, float *green, float *blue)
+{
+   double r = 3.2406 * x + (- 1.5372 * y) + (- 0.4986 * z);
```
Comment (SiegeLord) [inline addons/color/color.c]: What's up with the weird space after the negation here?

File: examples/ex_color2.c [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+}
+
+static void draw_range(int ci)
+{
+   float l = example.color[ci].l;
+   int cx = (example.color[ci].a * 511.0 / 2) + 511.0 / 2;
+   int cy = (-example.color[ci].b * 511.0 / 2) + 511.0 / 2;
+   int r = 2;
+   /*
+    * 1234561
```
Comment (SiegeLord) [inline examples/ex_color2.c]: Hah! What's this?

Comment (allefant) [inline examples/ex_color2.c]: I need to comment this. Basically we loop around the selected color in the center to find the pixels with a certain ciede2000 distance. We start around the center then go 1, 2 to the right. Then we go 1, 2 down. Then 1, 2 left. Then 1, 2 up. Then we jump to a bigger radius and start over with 1, 2, 3, 4 right, 1, 2, 3, 4 down... The diagram shows how exactly :)

File: addons/color/color.c (hunk 1 above)
Comment (allefant) [inline addons/color/color.c]: You clearly must not like Haskell :P

**Pull request merged**